presto-plugin-toolkit/src/main/java/com/facebook/presto/plugin/base/security/SystemAccessControlStatsRecorder.java

```diff
@@ -0,0 +1,723 @@
+    private final SystemAccessControlStats checkCanSetTableProperties = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanDropTable = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanRenameTable = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanShowTablesMetadata = new SystemAccessControlStats();
+    private final SystemAccessControlStats filterTables = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanShowColumnsMetadata = new SystemAccessControlStats();
+    private final SystemAccessControlStats filterColumns = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanAddColumn = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanDropColumn = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanRenameColumn = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanSelectFromColumns = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanInsertIntoTable = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanDeleteFromTable = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanTruncateTable = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanUpdateTableColumns = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanCreateView = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanRenameView = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanDropView = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanCreateViewWithSelectFromColumns = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanSetCatalogSessionProperty = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanGrantTablePrivilege = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanRevokeTablePrivilege = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanDropConstraint = new SystemAccessControlStats();
+    private final SystemAccessControlStats checkCanAddConstraint = new SystemAccessControlStats();
+    private final SystemAccessControlStats getRowFilters = new SystemAccessControlStats();
+    private final SystemAccessControlStats getColumnMasks = new SystemAccessControlStats();
+
+    public void recordCheckCanSetUser(long nanos)
+    {
+        checkCanSetUser.record(nanos);
+    }
+
+    public void recordCheckCanSetUserFailure()
+    {
+        checkCanSetUser.recordFailure();
+    }
+
+    public void recordSelectAuthorizedIdentity(long nanos)
+    {
+        selectAuthorizedIdentity.record(nanos);
+    }
```

Comment:
`recordXXX` methods like these are increasing code verbosity. As an inner class of `StatsRecordingSystemAccessControl`, keep the visibility of the fields such that you can call `checkCanSetUser.record(nanos);` at the call site